lib/Stats.js

```diff
@@ -118,37 +149,225 @@ class Stats {
 			compilation.compiler.context === context
 				? compilation.requestShortener
 				: new RequestShortener(context);
-		const showPerformance = optionOrLocalFallback(options.performance, true);
-		const showHash = optionOrLocalFallback(options.hash, true);
-		const showEnv = optionOrLocalFallback(options.env, false);
-		const showVersion = optionOrLocalFallback(options.version, true);
-		const showTimings = optionOrLocalFallback(options.timings, true);
-		const showBuiltAt = optionOrLocalFallback(options.builtAt, true);
-		const showAssets = optionOrLocalFallback(options.assets, true);
-		const showEntrypoints = optionOrLocalFallback(options.entrypoints, true);
+
+		const showCachedAssets = isTypeOf(options.cachedAssets, "boolean")
+			? options.cachedAssets
+			: options.all !== undefined
+				? options.all
+				: assertTypeOf(
+						this.hooks.setDefault.for("cachedAssets").call(),
+						"boolean"
+					);
+
+		const showAssets = isTypeOf(options.assets, "boolean")
+			? options.assets
+			: options.all !== undefined
+				? options.all
+				: assertTypeOf(this.hooks.setDefault.for("assets").call(), "boolean");
+
+		const sortAssets = isTypeOf(options.assetsSort, "string")
+			? options.assetsSort
+			: assertTypeOf(this.hooks.setDefault.for("assetsSort").call(), "string");
+
+		const excludeAssets = []
+			.concat(optionsOrFallback(options.excludeAssets, []))
+			.map(testAgainstGivenOption);
+
+		const showModules = isTypeOf(options.modules, "boolean")
+			? options.modules
+			: options.all !== undefined
+				? options.all
+				: assertTypeOf(this.hooks.setDefault.for("modules").call(), "boolean");
+
+		const showCachedModules = isTypeOf(options.cached, "boolean")
+			? options.cached
+			: options.all !== undefined
+				? options.all
+				: assertTypeOf(this.hooks.setDefault.for("cached").call(), "boolean");
+
+		const sortModules = isTypeOf(options.modulesSort, "string")
+			? options.modulesSort
+			: assertTypeOf(this.hooks.setDefault.for("modulesSort").call(), "string");
+
+		const showNestedModules = isTypeOf(options.nestedModules, "boolean")
+			? options.nestedModules
+			: options.all !== undefined
+				? options.all
+				: assertTypeOf(
+						this.hooks.setDefault.for("nestedModules").call(),
+						"boolean"
+					);
+
+		const showChildren = isTypeOf(options.children, "boolean")
+			? options.children
+			: options.all !== undefined
+				? options.all
+				: assertTypeOf(this.hooks.setDefault.for("children").call(), "boolean");
+
+		const showModuleTrace = isTypeOf(options.moduleTrace, "boolean")
```

Comment:
This pattern is repeated multiple times. It could be extracted to a function to improve the maintenance.

Reply:
Agreed, on it